gadget/raw.go

```diff
@@ -292,7 +303,12 @@ func (r *RawStructureUpdater) Update() error {
 	}
 	defer disk.Close()
 
-	for _, pc := range r.ps.PositionedContent {
+	structForDevice := *r.ps
+	if offs != r.ps.StartOffset {
+		structForDevice = ShiftStructureTo(*r.ps, offs)
+	}
```

Comment:
Super nitpick, since structForDevice may get thrown away immediately if offs != r.ps.StartOffset, how about:
```
var structForDevice PositionedStructure
if offs != r.ps.StartOffset {
  structForDevice = ShiftStructureTo(*r.ps, offs)
} else {
  structForDevice = *r.ps
}
```
?
 Also applies to similiar assignment earlier in the code.

Reply:
Refactored into a separate helper. Please take a look.

Reply:
The helper looks very nice, thanks!